<h2>Promise</h2>

<pre><code class="js">
Promise.resolve(55)
    .then((val) => {
        console.log(val);
        throw new Error();
    })
    .then((val) => console.log('then', val))
    .catch((val) => {
        console.log('catch', val);
        return Promise.reject();
    })
    .then(finalHandler, finalHandler1)
    .then(finalHandler, finalHandler1)
    .then(finalHandler, finalHandler1);

function finalHandler() {
    console.log('finalHandler')
}
function finalHandler1() {
    console.log('finalHandler1')
}
// => 55, catch Error, finalHandler1, finalHandler, finalHandler
</code></pre>
